Finalized play playlist use case

rafaelgacesa opened the pull request:
### In this pull:
- Files required for the play playlist use case
   - uses synchronization to play a queue of songs
   - uses play song as an output boundary so no output boundary needed
- A dummy implementation of some play song stuff
   - @Mistryr222 will probably have to change a lot of it, but feel free to reuse anything useful

I had to recreate this pull request due to some weird Git issues. Hopefully the autograder passes now.

sarxhlwalker reviewed (approved):
Good thinking on the threads!

rafaelgacesa commented:
Added unit tests and modified `playNext()`'s handling of `stopQueue()` slightly. Autograder will fail the tests because it can't find the audio files for some reason, but they pass locally:

<img width="314" alt="Screenshot 2022-11-11 at 10 35 44 PM" src="https://user-images.githubusercontent.com/68714873/201454504-25c73b4e-c483-4808-ae37-0ef6521200d3.png">

chrstinalin commented on src/main/java/UseCase/playPlaylist.java:
```diff
@@ -1,4 +1,72 @@
+    /**
+     * Dequeues the playlist, such that after the currently playing song ends the playlist
+     * will not continue. Must be called whenever audio source is switched (eg. user plays new song,
+     * different playlist, or space).
```
When someone clicks 'play' on a new song in the middle of a playPlaylist, is this insinuating that the current song must /finish/ playing before the song a user wanted plays?

rafaelgacesa replied:
Nope! When someone calls play() again on the Music player, the current player is immediately killed. This method is here to prevent the _next_ song of the Playlist randomly interrupting the new content that was clicked on.

chrstinalin commented on src/test/java/UseCase/PPlaybackTest.java:
```diff
@@ -0,0 +1,90 @@
+    public void testPlay(){
+        File file = new File("./src/test/java/UseCase/test1.mp3");
+        Song song = new Song(0, null, null, 0, null, file, false, null);
+        File file2 = new File("./src/test/java/UseCase/test2.mp3");
```
Assuming these are temporary?

rafaelgacesa replied:
They can be, but I figured it doesn't really matter since it's the test folder and not the main one.

chrstinalin reviewed (dismissed):
Looks good, interesting stuff. Just the two clarification comments, nothing major.

sarxhlwalker reviewed (dismissed):
Looks good. One question about the tests -- assuming we merge these into main, will all of our autograde GitHub tests fail in the future? (Is the plan to continuously show screenshot proof of the tests passing?)

rafaelgacesa commented:
> Looks good. One question about the tests -- assuming we merge these into main, will all of our autograde GitHub tests fail in the future? (Is the plan to continuously show screenshot proof of the tests passing?)

That is a great question lol. I'll comment the assertions out with a small note when I merge this.

sarxhlwalker reviewed (approved):
we love merge conflicts

Mistryr222 reviewed (approved):
I added to some of these files in my PR but I made sure that they're similar enough so that even if there's a merge conflict with mine after its a small change. Use case looks good.

rafaelgacesa commented on src/main/java/UseCase/playSong.java:
```diff
@@ -1,13 +1,26 @@
+        this.song = s.getSong();
+    }
+
+    public static void playAudio(File song){
```
Yep, that’s temporary.

NicholasMacasaet replied:
Ok cool, as long as it gets implemented at some point I don't see any issues with this.

chrstinalin reviewed (approved):
Pretty sure my next pull will have conflicts too that'll be fun 😃
